File: unit-2/sprint-2/day-3-RecyclerView-II/assignments/ID_CARD Continue/app/src/main/java/com/example/idcardcontinue/MainActivity.java
package com.example.idcardcontinue;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.content.DialogInterface;
import android.os.Bundle;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements ItemClickListener {
    private RecyclerView recyclerView;
    private ArrayList<Student> students;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        intiViews();
        buildRecyclerData();
        setRecyclerView();
    }

    private void setRecyclerView() {
        StudentAdaptor studentAdaptor = new StudentAdaptor(students, this);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(linearLayoutManager);
        recyclerView.setAdapter(studentAdaptor);
    }

    private void buildRecyclerData() {
        students = new ArrayList<>();
        for (int i = 0; i < 100; i++) {
            Student student1 = new Student("Microsoft", 64, "Business", R.drawable.jeff_bezos);
            students.add(student1);

        }
    }

    private void intiViews() {
        recyclerView = findViewById(R.id.recyclerView);
    }

    @Override
    public void onClick(Student student, int adapterPosition) {
        AlertDialog dialog = new AlertDialog.Builder(this)
                .setTitle("Hey this is the Data")
                .setMessage(
                        student.getCompanyName() + "\n"
                                + student.getAge() + "\n"
                ).
                        setPositiveButton("Ok", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {
                                dialog.dismiss();
                            }
                        }).
                        setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {
                                dialog.dismiss();
                            }
                        }).
                        show();
    }
}

File: unit-5/data binding/Test/app/src/main/java/com/masai/myapplication/ui/views/MainActivity.kt
package com.masai.myapplication.ui.views

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.masai.myapplication.R
import com.masai.myapplication.data.models.ResultDTO
import com.masai.myapplication.ui.adapter.MainAdapter
import com.masai.myapplication.ui.viewmodel.MainViewModel
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch


class MainActivity : AppCompatActivity() {
    private lateinit var mainViewModel: MainViewModel
    var list = ArrayList<ResultDTO>()
lateinit var mainAdapter : MainAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mainViewModel = ViewModelProvider(this).get(MainViewModel::class.java)
        setRecyclerAdapter()
        mainViewModel.callAPi()

        mainViewModel.liveData.observe(this,{
            list.clear()
            list.addAll(it)
            mainAdapter.notifyDataSetChanged()
        })
    }
    fun setRecyclerAdapter() {
         mainAdapter = MainAdapter(list)
        val linearLayoutManager = LinearLayoutManager(this)
        recyclerView.apply {
            adapter = mainAdapter
            layoutManager = linearLayoutManager
        }
    }
}